krpc-code-gen/src/main/java/io/kroxylicious/krpccodegen/main/KrpcGenerator.java

```diff
@@ -200,7 +223,46 @@ private void render(Configuration cfg, MessageSpec messageSpec) {
         });
     }
 
-    private Stream<MessageSpec> messageSpecs() {
+    private void renderMulti(Configuration cfg, Set<MessageSpec> messageSpecs) {
+        logger.log(Level.INFO, "Processing message specs");
+
+        // TODO not actually used right now
+        // var structRegistry = new StructRegistry();
+        // try {
+        // for (MessageSpec messageSpec : messageSpecs) {
+        // structRegistry.register(messageSpec);
+        // }
+        // }
+        // catch (Exception e) {
+        // throw new RuntimeException(e);
+        // }
+        templateNames.forEach(templateName -> {
+            try {
+                logger.log(Level.DEBUG, "Parsing template {0}", templateName);
+                var template = cfg.getTemplate(templateName);
+                // TODO support output to stdout via `-`
+                var outputFile = new File(outputDir, outputFile(outputFilePattern, null, templateName));
+                logger.log(Level.DEBUG, "Opening output file {0}", outputFile);
+                try (var writer = new OutputStreamWriter(new FileOutputStream(outputFile), outputEncoding)) {
+                    Map<String, Object> dataModel = Map.of(
+                            // "structRegistry", structRegistry,
+                            "outputPackage", outputPackage,
+                            "messageSpecs", messageSpecs,
+                            "toSnakeCase", new SnakeCase(),
```

Comment:
@tombentley, do we actually need this? Is it obsoleted by that new `RetrieveApiKey` function below?

Reply:
`grep` knows the answer better than me!